API/projectClass/customFileClass.m

```diff
@@ -74,26 +75,39 @@ function delete(obj)
                             throw(exceptions.invalidType('Single input is expected to be a custom object name'));
                         end
                         
-                        newRow = {newName, '', supportedLanguages.Matlab.value, ''};
+                        newRow = {newName, '', '', supportedLanguages.Matlab.value, ''};
                         
                     case 2
 
                         % Two inputs suppled - assume both name and
                         % filename supplied;
                         newName = char(inputs{1});
                         newFile = char(inputs{2});
+                        [~, functionName, ~] = fileparts(newFile);
 
-                        newRow = {newName, newFile, supportedLanguages.Matlab.value, ''};
+                        newRow = {newName, newFile, functionName, supportedLanguages.Matlab.value, ''};
                         
                     case 4
 
-                        % Four inputs = assume all inputs supplied
+                        % Four inputs = assume all inputs except function name supplied
```

Comment:
I think this design where argument 3 changes from **function name** to **language** will be confusing to some users. Its always best to add optional args at the end so we could either remove the number 4 case or move function name to the end
```
customFiles.addCustomFile('New Row', 'file.py', 'python', 'C:/stuff')
customFiles.addCustomFile('New Row', 'file.py', 'py_function', 'python', 'C:/stuff')
```

Reply:
Yeah, that sounds good. I did want to have the function name next to the file name though, but it's probably best to just put it on the end and have everything the same in that case.

Reply:
Ok, let's do that and see how it goes.